Mono.Debugger.Soft/Mono.Debugger.Soft/VirtualMachine.cs

```diff
@@ -198,6 +197,7 @@ internal void AddThreadToInvalidateList (ThreadMirror threadMirror)
 				//if (threadCache != threads) {//While fetching threads threadCache was invalidated(thread was created/destoyed)
 				//	return GetThreads ();
 				//}
+				threadCache = threads;
```

Comment:
I think technically you need a call to `Thread.MemoryBarrier()` before this assignment so that all the writes to the `threads` array are seen by other threads.  Otherwise they may still see null.  (I guess we only ever run this code on ARM on Mono's CI, but still...)

Reply:
Done